[Jordan Chan] iP

naythee169 opened the pull request:
## Delphi
> "Be yourself; everyone else is already taken" - Oscar Wilde ([source](https://blog.hubspot.com/sales/famous-quotes)).

Delphi is the best AI personal assistant on the market today. It's
- text-based
- understands **many** different types of input 
- friendly!

All you need to do is,

1. Download it from ([here](https://github.com/naythee169/ip)).
2. open it
3. run it

And that's it! easy peasy lemon squeezy 👍 .

Features:

- [x] store different types of tasks
- [x] search for tasks by name
- [ ] reorder tasks by priority(coming soon)

If you are interested, here's the `run()` method that helps run the bot:

```Public class Delphi {
public void run() {
        boolean isExitCommand = false;
        ui.welcomeMessage();
        System.out.println("current tasks:");
        taskList.loadStorageToTasks(this.storage);
        taskList.printTasks();
        Scanner scanner = new Scanner(System.in);
        while (!isExitCommand) {
            try {
                Command c = parser.parseInput(scanner.nextLine());
                c.execute(taskList, storage, ui);
                isExitCommand = c.exitBot();
            } catch (DelphiException e) {
                System.out.println(e.getMessage());
            }
        }
        ui.goodbyeMessage();
    }
}

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

naythee169 committed:
Add rename(to Delphi), greet and exit

naythee169 committed:
Add echo, new personality

naythee169 committed:
Add list, add functions

naythee169 committed:
Add mark as done function

naythee169 committed:
Change toString() functions to align more with OOP

naythee169 committed:
Create separate class for string parser

naythee169 committed:
Add todo, event, deadline

naythee169 committed:
Update automated text UI testing

naythee169 committed:
Add exception handling for empty and invalid inputs

naythee169 committed:
Add delete tasks function

naythee169 committed:
Fix minor copy-paste errors in TaskList

naythee169 committed:
Align with java coding standard

naythee169 committed:
Add javadocs comments to TaskList

naythee169 committed:
Add InvalidListItemException

naythee169 committed:
Add Javadocs comments for Parser

naythee169 committed:
Split Task subtypes into separate class files

naythee169 committed:
Modify previous Task subtype files

naythee169 committed:
Add hard disk functionality

naythee169 committed:
Merge branch 'branch-Level-7' with hard disk functionality

naythee169 committed:
Add date/time functionality

naythee169 committed:
Merge branch 'branch-Level-8' to add date/time functionality

naythee169 committed:
Implement more OOP design, including UI and Storage classes

naythee169 committed:
Separate class files into packages

naythee169 committed:
Merge branch 'branch-A-Packages' which splits the class files into packages

naythee169 committed:
Merge remote-tracking branch 'origin/add-gradle-support'

naythee169 committed:
Configure gradle and created more packages

naythee169 committed:
Add Commands package for greater OOP adherence

naythee169 committed:
Add JUnit tests

naythee169 committed:
Create JAR file

naythee169 committed:
Add JavaDoc comments

naythee169 committed:
Align with coding standards

naythee169 committed:
Add find feature

naythee169 committed:
Merge branch 'branch-A-JavaDoc'

naythee169 committed:
Merge branch 'branch-A-CodingStandard'

naythee169 committed:
Merge branch 'branch-Level-9'

naythee169 committed:
Import find command into Parser

naythee169 committed:
Fix coding style errors that were missed previously

naythee169 committed:
Update welcome message

naythee169 committed:
Fix more style violations in test cases

Kitty-001 reviewed (commented):
LGTM.
Code is clean and nice. Everything is perfect to me 😭

naythee169 committed:
Add GUI

naythee169 committed:
Merge branch 'branch-Level-10'

bbryant824 commented on src/main/java/UI/Delphi.java:
```diff
@@ -0,0 +1,86 @@
+     * Starts the application, handles user input, and manages the task list.
+     * Displays the welcome message, processes commands, and exits upon receiving the exit command.
+     */
+    public void run() {
```
used appropriate names for your variables and methods. For example, taskList, storage, and parser are clear noun-based variable names representing things, and method names like run() and getResponse() follow the verb-based naming convention.

bbryant824 commented on src/main/java/UI/Delphi.java:
```diff
@@ -0,0 +1,86 @@
+     *
+     * @return the tasks stored in the hard drive at the time of starting the bot
+     */
+    public String currentTasks() {
```
The method currentTasks() would benefit from a more descriptive verb-based name, such as getCurrentTasks(), to indicate that the method retrieves the current tasks rather than performing an action.

bbryant824 commented on src/main/java/UI/Delphi.java:
```diff
@@ -0,0 +1,86 @@
+     */
+    public String currentTasks() {
+        taskList.loadStorageToTasks(this.storage);
+        String res = "current tasks: \n" + taskList.printTasks();
```
Replace ambiguous variables like res with more descriptive names such as taskSummary to improve clarity.

bbryant824 commented on src/main/java/UI/DialogBox.java:
```diff
@@ -0,0 +1,62 @@
+ * Represents a dialog box consisting of an ImageView to represent the speaker's face
+ * and a label containing text from the speaker.
+ */
+public class DialogBox extends HBox {
```
Appropriately used nouns for your classes (DialogBox, Label, ImageView) and methods that represent actions, such as getUserDialog() and getDukeDialog(). The naming is clear and descriptive, which enhances code readability.

bbryant824 commented on src/main/java/UI/DialogBox.java:
```diff
@@ -0,0 +1,62 @@
+    /**
+     * Flips the dialog box such that the ImageView is on the left and text on the right.
+     */
+    private void flip() {
```
The method flip() could be renamed to something more descriptive like flipDialogBox() to better indicate the action it performs, as flip() on its own might not fully convey the context to someone unfamiliar with the class.

bbryant824 reviewed (commented):
You’ve mostly followed good naming conventions, using nouns for objects (dialog, displayPicture) and verbs for actions (getUserDialog(), getDukeDialog()). The class and method names are descriptive and intuitive, which improves code readability. Very welldone, can see you also added css to improve the UI design, respect the efforts!!

naythee169 committed:
Align with coding quality standards

naythee169 committed:
Add assertions

naythee169 committed:
Merge pull request #1 from naythee169/branch-A-CodeQuality

Align with coding quality standards

naythee169 committed:
Merge pull request #2 from naythee169/branch-A-Assertions

Add assertions

naythee169 committed:
Add BCD extension

Currently, users can only add new tasks but cannot modify existing ones
without deleting them.

A more flexible task feature is required to let users modify the
existing tasks.

We add an extension to introduce the ability to modify the deadline
associated with the deadline in Deadline tasks and time frame
for Event tasks by adding an edit task method to the Task class.

This is so that OOP principles are adhered to still.

naythee169 committed:
Merge branch 'master' of https://github.com/naythee169/ip

naythee169 committed:
Modify test cases

Current test cases do not work due to changes to the Event and Deadline
classes when implementing the extension

The test cases for Event need to change to be appropriate in verifying
the Event behaviour

Change the Parser stub so that it now produces the desired behaviour
when used by the Event class. Modify the test cases to not throw
errors anymore.

This is the best way to get the test cases to work properly

naythee169 committed:
Improve code quality and style

naythee169 committed:
Align with checkstyle requirements

naythee169 committed:
upgrade GUI

naythee169 committed:
Add more robust error handling

naythee169 committed:
Add more robust JUnit testing

naythee169 committed:
Add representative screenshot

naythee169 committed:
Add README file

naythee169 committed:
Update correct README file

naythee169 committed:
Improve code quality and OOP

naythee169 committed:
Fix README file bug where images weren't rendering

naythee169 committed:
Update README file to add missing command

naythee169 committed:
Fix checkstyle violations

naythee169 committed:
Tweak README file for tagging purposes

naythee169 committed:
Fix bug in storage class

naythee169 committed:
Improve error message

naythee169 committed:
Add credit to another student for reused code

naythee169 committed:
Tweak list comand

naythee169 committed:
Add link to jar file in README
